rand_distr/src/normal_inverse_gaussian.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{Distribution, Float, InverseGaussian, Standard, StandardNormal};
+use rand::Rng;
+
+/// Error type returned from `NormalInverseGaussian::new`
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// `alpha <= 0` or `nan`.
+    AlphaNegativeOrNull,
+    /// `|beta| >= alpha` or `nan`.
+    AbsoluteBetaLessThanAlpha,
+}
+
+/// The [normal-inverse Gaussian distribution](https://en.wikipedia.org/wiki/Normal-inverse_Gaussian_distribution)
+#[derive(Debug)]
+pub struct NormalInverseGaussian<N> {
+    alpha: N,
+    beta: N,
+    inverse_gaussian: InverseGaussian<N>,
+}
+
+impl<N: Float> NormalInverseGaussian<N>
+where StandardNormal: Distribution<N>
+{
+    /// Construct a new `NormalInverseGaussian` distribution with the given alpha (tail heaviness) and
+    /// beta (asymmetry) parameters.
+    pub fn new(alpha: N, beta: N) -> Result<NormalInverseGaussian<N>, Error> {
+        if !(alpha > N::from(0.0)) {
+            return Err(Error::AlphaNegativeOrNull);
+        }
+
+        if !(beta.abs() < alpha) {
+            return Err(Error::AbsoluteBetaLessThanAlpha);
+        }
```

Comment:
Since you're calculating `α² - β²` anyway, it may be better to check that `γ > 0`.

Reply:
I need to check that `|β| < α` otherwise `γ` would be the square root of a negative number

Reply:
Sorry, I meant `α² - β² > 0`, but it doesn't make much difference anyway.